libraries/AP_Baro/AP_Baro_UAVCAN.cpp

```diff
@@ -31,23 +26,25 @@ void AP_Baro_UAVCAN::subscribe_msgs(AP_UAVCAN* ap_uavcan)
     if (ap_uavcan == nullptr) {
         return;
     }
-
-    auto* node = ap_uavcan->get_node();
-
-    uavcan::Subscriber<uavcan::equipment::air_data::StaticPressure, PressureCb> *pressure_listener;
-    pressure_listener = new uavcan::Subscriber<uavcan::equipment::air_data::StaticPressure, PressureCb>(*node);
-    // Msg Handler
-    const int pressure_listener_res = pressure_listener->start(PressureCb(ap_uavcan, &handle_pressure));
-    if (pressure_listener_res < 0) {
-        AP_HAL::panic("UAVCAN Baro subscriber start problem\n\r");
+#if AP_TEST_DRONECAN_DRIVERS
+    get_uavcan_backend(ap_uavcan, 125, true); // add entry for test driver
+#endif
+    auto *pressure_cb = Canard::allocate_arg_callback(ap_uavcan, &handle_pressure);
```

Comment:
why is this two calls, first to allocate_arg_callback() then to Canard::Subscriber. That means two lots of allocation failure handling.

Reply:
I guess I can add another template argument of Callback type to Canard::Subscriber<> if you want.